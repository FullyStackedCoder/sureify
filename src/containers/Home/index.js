import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import * as actions from '../../store/actions';
import Spinner from '../../components/Spinner';
import Table from '../../components/Table';

const Home = props => {
  useEffect(() => {
    props.onFetchHomeData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleSort = (event, id) => {
    console.log(event.target, id);
    props.onSortData(id);
  };

  const handleFilter = event => {
    console.log(event.target.value);
    props.onFilterData(event.target.value);
  };

  if (props.loading) {
    return <Spinner />;
  }

  return (
    <main className="container">
      <div className="search">
        <input
          type="text"
          placeholder="Search"
          id="search"
          onChange={handleFilter}
          value={props.searchInput}
        />
      </div>
      <div className="strategies">
        <Table
          data={
            props.searchInput && props.searchInput.length > 0
              ? props.searchResults
              : props.data
          }
          click={handleSort}
          searchInput={props.searchInput}
        />
      </div>
    </main>
  );
};

const mapStateToProps = state => {
  return {
    loading: state.home.loading,
    error: state.home.error,
    data: state.home.data,
    searchInput: state.home.searchInput,
    searchResults: state.home.searchResults
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onFetchHomeData: () => dispatch(actions.fetchHomeData()),
    onSortData: id => dispatch(actions.sortData(id)),
    onFilterData: searchInput => dispatch(actions.filterData(searchInput))
  };
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Home));
