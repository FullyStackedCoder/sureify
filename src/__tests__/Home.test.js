import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import Home from '../containers/Home';
import '../setupTests';

describe('<Home />', () => {
  it('should render the Home page <Home /> correctly', () => {
    const mockStore = configureStore();
    const store = mockStore({
      home: { data: fakeData, error: null, loading: false }
    });
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

const fakeData = {
  columnHeaders: [
    {
      id: 'firstName',
      label: 'First Name',
      sortable: 'true',
      filterable: 'true'
    },
    {
      id: 'lastName',
      label: 'Last Name',
      sortable: 'true',
      filterable: 'true'
    },
    {
      id: 'dateOfBirth',
      label: 'Date of Birth',
      sortable: 'false',
      filterable: 'true'
    },
    {
      id: 'designation',
      label: 'Designation',
      sortable: 'true',
      filterable: 'true'
    },
    {
      id: 'department',
      label: 'Department',
      sortable: 'true',
      filterable: 'true'
    }
  ],
  rowData: [
    {
      id: 'user_id_1',
      data: [
        {
          id: 'firstName',
          value: 'John'
        },
        {
          id: 'lastName',
          value: 'Smith'
        },
        {
          id: 'dateOfBirth',
          value: '07-11-1992'
        },
        {
          id: 'designation',
          value: 'Software Engineer'
        },
        {
          id: 'department',
          value: 'Web'
        }
      ]
    },
    {
      id: 'user_id_2',
      data: [
        {
          id: 'firstName',
          value: 'Jane'
        },
        {
          id: 'lastName',
          value: 'Doe'
        },
        {
          id: 'dateOfBirth',
          value: '07-11-1995'
        },
        {
          id: 'designation',
          value: 'Intern'
        },
        {
          id: 'department',
          value: 'Mobile'
        }
      ]
    }
  ]
};
