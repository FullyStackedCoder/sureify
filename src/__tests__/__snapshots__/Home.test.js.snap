// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the Home page <Home /> correctly 1`] = `ReactWrapper {}`;
